src/ray/common/scheduling/cluster_resource_scheduler.cc

```diff
@@ -388,13 +391,13 @@ void ClusterResourceScheduler::ResourceMapToTaskRequest(
     } else if (it->first == ray::kMemory_ResourceLabel) {
       task_request->predefined_resources[MEM].demand = it->second;
     } else {
-      // This is a custom resource.
       task_request->custom_resources[i].id = string_to_int_map_.Insert(it->first);
       task_request->custom_resources[i].req.demand = it->second;
       task_request->custom_resources[i].req.soft = false;
       i++;
     }
   }
+  task_request->custom_resources.resize(i);
```

Comment:
Why do we need to resize after every loop iteration? Won't this just end up the same as `task_request->custom_resources.resize(resource_map.size());`?